File: backend/src/main/java/com/haruma/library/entity/Order.java
package com.haruma.library.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.util.Date;

@Entity
@Table(name="`order`")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
@Builder
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="order_id")
    private Long orderId;

    @Column(name="order_date")
    private Date orderDate;

    @Column(name="total_price")
    private BigDecimal totalPrice;

    @Column(name="quantity", nullable = false)
    private Integer quantity;

    @ManyToOne(cascade = {CascadeType.DETACH, CascadeType.MERGE, CascadeType.PERSIST, CascadeType.REFRESH}, optional = false)
    @JoinColumn(name = "user_id")
    @JsonIgnore
    private User user;

    @ManyToOne(cascade = {CascadeType.DETACH, CascadeType.MERGE, CascadeType.PERSIST, CascadeType.REFRESH}, optional = false)
    @JoinColumn(name = "status_id", referencedColumnName = "status_id", nullable = false)
    private Status status;


    @ManyToOne(cascade = {CascadeType.DETACH, CascadeType.MERGE, CascadeType.PERSIST, CascadeType.REFRESH}, optional = false)
    @JoinColumn(name = "address_id", referencedColumnName = "address_id", nullable = false)
    private Address address;


    @ManyToOne(cascade = {CascadeType.DETACH, CascadeType.MERGE, CascadeType.PERSIST, CascadeType.REFRESH})
    @JoinColumn(name = "book_id", referencedColumnName = "id", nullable = false)
    private Book book;
}

File: frontend/src/model/Order.ts
import Address from './Address';
import Book from './Book';
import Status from './Status';

class Order {
	public orderId: number;
	public orderDate: number;
	public totalPrice: number;
	public quantity: number;
	public status: Status;
	public address: Address;
	public book: Book;
	public constructor(
		orderId: number,
		orderDate: number,
		totalPrice: number,
		quantity: number,
		status: Status,
		address: Address,
		book: Book,
	) {
		this.orderId = orderId;
		this.orderDate = orderDate;
		this.address = address;
		this.totalPrice = totalPrice;
		this.status = status;
		this.book = book;
		this.quantity = quantity;
	}
}

export default Order;
